jerry-core/ecma/builtin-objects/ecma-builtin-helpers-date.c

```diff
@@ -333,7 +333,7 @@ ecma_date_daylight_saving_ta (jerry_time_zone_t *tz, /**< time zone information
    * TODO: Fix daylight saving calculation.
    *       https://github.com/jerryscript-project/jerryscript/issues/1661
    */
-  return tz->daylight_saving_time * ECMA_DATE_MS_PER_HOUR;
+  return (ecma_number_t) tz->daylight_saving_time * ECMA_DATE_MS_PER_HOUR;
```

Comment:
Perhaps it would no be allowed to use the date with float32 since even the current date can not be represented.